### DISCUSSION TITLE
How can I extract gene annotations from BSgenome?

### DISCUSSION THREAD

**Discussion by qingzengLi** [stuart-lab/signac#589] [Q&A] [answered]
I'm a novice. How can I extract gene annotations from BSgenome? For exampe, I want to add gene annotation information from BSgenome.Athaliana.TAIR.04232008

**Comment by timoast** [comment, marked as answer]
Hi @qingzengLi 

The BSgenome object does not contain gene annotations, you will need a different object (EnsDb gene annotations, or a gff/gtf file). Please see the vignettes available on the Signac website

**Reply by qingzengLi** [reply]
thank you!
I try to extract gene annotations from gtf file, but encountered some problems when executing TSSEnrichment().here is my code.

gtf <- rtracklayer::import("F:\scATAC_scSeq\Arabidopsis_thaliana.TAIR10.50.gtf")
gene.coords <- gtf[gtf$type == 'gene']
seqlevelsStyle(gene.coords) <- 'Ensembl'
gene.coords <- keepStandardChromosomes(gene.coords,pruning.mode = 'coarse')
Annotation(ATAC1) <- gene.coords
ATAC1 <- TSSEnrichment(object = ATAC1,fast = FALSE)

Extracting TSS positions
Error in eval(bysub, x, parent.frame()) : object not found 'gene_name'
